### PR TITLE
Dockerize sync tool.

### PR DESCRIPTION
**Pull request by wlynch**
# Changes

This achieves a few things:

1. Creates Dockerfile for sync tool, so that it can be invoked from
   Tekton tasks.
2. Makes sync config directory configurable by flag, so that the tool
   can be ran from any directory. This defaults to sync/config (same as
   before).
3. Moves requirements.txt to sync directory to be closer to actual code.
   Since we're not going to be using Netlify Build for deployments,
   placement of this file doesn't matter.

Issue: #34

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/website/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: sync/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+WORKDIR /app
+# Only copy specific values instead of the entire directory.
+# We're primarily trying to avoid pulling in config/, since it's confusing that
+# it will not be used.
+COPY sync.py sync.py
+COPY requirements.txt requirements.txt
+
+RUN pip install -r requirements.txt
+
+ENTRYPOINT [ "python3", "sync.py" ]
```
Comment (stevemar) [inline sync/Dockerfile:12]: you can just use `python` here as you're pulling the python3.7 image...

```
$ docker run -it python:3.7 bash
root@6989d0bbf8d2:/# which python
/usr/local/bin/python
root@6989d0bbf8d2:/# python --version
Python 3.7.7
```

**Comment by wlynch** [comment]
> why not delete the top level `requirements.txt` and `runtime.txt` files, too?

It's a requirement of Netlify Build. https://docs.netlify.com/configure-builds/manage-dependencies/#python

Updated the commit message to include this!

File: sync/Dockerfile (hunk 1 above)
Comment (wlynch) [reply sync/Dockerfile:12]: While this is also correct, I'd prefer to keep this as python3.

I find that I use Dockerfiles as a guide for running applications locally for development and testing, and we can't assume that the default system python is necessarily python3 (particularly on OSX).

On the image `python` is just a symlink to `python3`, so using either is the same.

```
$ docker run -it python:3.7 bash
root@58befeb9dc6c:/# ls -al $(which python)
lrwxrwxrwx 1 root root 7 May 16 07:18 /usr/local/bin/python -> python3
```

**Comment by stevemar** [comment]
> It's a requirement of Netlify Build. https://docs.netlify.com/configure-builds/manage-dependencies/#python

Ahh, sneaky! I would recommend adding this specific bit of into to https://github.com/tektoncd/website/blob/master/DEVELOPMENT.md but this PR is fine as-is.

/lgtm

**Pull request merged**